show-line-of-sight-between-geoelements/src/main/java/com/esri/arcgismaps/sample/showlineofsightbetweengeoelements/screens/MainScreen.kt

```diff
@@ -0,0 +1,103 @@
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.width
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Slider
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.collectAsState
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.dp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import com.arcgismaps.toolkit.geoviewcompose.SceneView
+import com.esri.arcgismaps.sample.sampleslib.components.SampleTopAppBar
+import com.esri.arcgismaps.sample.showlineofsightbetweengeoelements.components.SceneViewModel
+
+/**
+ * Main screen layout for the sample app
+ */
+@Composable
+fun MainScreen(sampleName: String) {
+
+    // Define the viewmodel of this sample
+    val sceneViewModel: SceneViewModel = viewModel()
+
+    // Retrieve any changes to the z value from SceneViewModel
+    val currentZValue = sceneViewModel.currentZValue.collectAsState().value
+
+    // Defined in order to keep the z value in the positive range
+    val offset = 100
+
+    Scaffold(topBar = { SampleTopAppBar(title = sampleName) },
+        content = {
+            Column {
+                Box(
+                    modifier = Modifier
+                        .fillMaxSize()
+                        .weight(1f)
+                ) {
+                    // Composable function that wraps the SceneView
+                    SceneView(
+                        modifier = Modifier
+                            .fillMaxSize()
+                            .padding(it),
+                        arcGISScene = sceneViewModel.scene,
+                        analysisOverlays = listOf(sceneViewModel.analysisOverlay),
+                        graphicsOverlays = listOf(sceneViewModel.graphicsOverlay),
+                    )
+
+                    // Composable function that holds the slider and the text position value
+                    Row(
+                        modifier = Modifier.Companion
+                            .align(Alignment.BottomCenter)
+                            .fillMaxWidth()
+                            .background(Color.White),
+                        verticalAlignment = Alignment.CenterVertically,
+                        horizontalArrangement = Arrangement.SpaceBetween
+                    ) {
+
+                        Slider(
+                            value =( currentZValue.toFloat() + offset),
+                            onValueChange = { newHeight ->
+                                sceneViewModel.updateHeight(newHeight.toDouble() - offset)
+                            },
+                            valueRange = 0f..300f,
+                            modifier = Modifier
+                                .width(300.dp)
```

Comment:
If the Row contains significant horizontal padding, you can get away with using a fixed width. 
```suggestion
```